core/src/main/java/tc/oc/pgm/scoreboard/SidebarMatchModule.java

```diff
@@ -466,48 +473,59 @@ private void renderSidebar() {
       }
 
       for (Competitor competitor : sortedCompetitors) {
-        if (!firstTeam) {
+        if (!(firstTeam || isSuperCompactWool)) {
           // Add a blank row between teams
           rows.add("");
         }
         firstTeam = false;
 
-        // Add a row for the team name
-        rows.add(competitor.getStyledName(NameStyle.FANCY).render().toLegacyText());
-
-        if (isCompactWool()) {
-          String woolText = " ";
-          boolean firstWool = true;
-
-          List<Goal> sortedWools = new ArrayList<>(gmm.getGoals(competitor));
-          Collections.sort(
-              sortedWools,
-              new Comparator<Goal>() {
-                @Override
-                public int compare(Goal a, Goal b) {
-                  return a.getName().compareToIgnoreCase(b.getName());
+        // Prevent team name from showing up if there is no space for its objectives
+        if (!isSuperCompactWool || rows.size() + 2 < MAX_ROWS) {
```

Comment:
This condition is better handled above (before potentially adding the whitespace line), also prevents the whole code getting indented inwards by 1